src/web/lib/updateRelativeDates.ts

```diff
@@ -0,0 +1,66 @@
+import { makeRelativeDate } from '@root/src/web/lib/dateTime';
+
+const DEFAULT_INTERVAL = 15 * 1000; // 15 seconds
+
+type Options = {
+	interval?: number;
+};
+
+export const updateRelativeDates = (options?: Options): number[] => {
+	if (typeof window === 'undefined') return [];
+	const intervalIds: number[] = [];
+	document
+		.querySelectorAll('time[data-relativeformat]')
+		.forEach((element) => {
+			if (element instanceof HTMLElement) {
+				// Get required values
+				const { relativeformat: relativeFormat } = element.dataset;
+				const absoluteTime = element.getAttribute('datetime');
+				if (!absoluteTime || !relativeFormat) return [];
+
+				// Decide what interval to use
+				const elementInterval =
+					element.dataset.interval &&
+					parseInt(element.dataset.interval, 10);
+				const interval: number =
+					elementInterval || options?.interval || DEFAULT_INTERVAL;
+
+				// Create a new interval for this element
+				intervalIds.push(
+					window.setInterval(() => {
+						let newTime;
+						switch (relativeFormat) {
```

Comment:
should we support default?

Reply:
> I would prefer to just have a TS file that would return a number/string and that being used in JSX

I thought about that but the limitation of a doing this via jsx is I think you'd need all content hydrated? We would also not be supporting the existing contract that's in place for Frontend which is nice to do because then dates stay updated, even if the `time` elements land on the page via embeds or interatives.  Like this, so long as the contract is met, it works.

Reply:
> should we support default?

In this case I think ts prevents us ending up here with any other option. In any case, there's a [replacement function](https://github.com/guardian/libs/pull/134) for `makeRelativeDates` in the pipeline which will remove this model.